packages/advanced-logic/src/extensions/payment-network/erc777/stream.ts

```diff
@@ -36,64 +36,171 @@ export default class Erc777StreamPaymentNetwork<
+      }
+
+      if (!creationParameters.expectedStartDate) {
+        throw Error('expectedStartDate should not be empty');
+      }
+
+      return super.createCreationAction(
+        creationParameters,
+      ) as ExtensionTypes.IAction<TCreationParameters>;
     }
 
-    if (!creationParameters.expectedStartDate) {
-      throw Error('expectedStartDate should not be empty');
+    /* Subsequent Request Creation */
+    if (
+      creationParameters.masterRequestId &&
+      creationParameters.previousRequestId &&
+      creationParameters.recurrenceNumber
+    ) {
+      if (
+        (creationParameters.masterRequestId === creationParameters.previousRequestId &&
+          creationParameters.recurrenceNumber !== 1) ||
+        (creationParameters.recurrenceNumber === 1 &&
+          creationParameters.masterRequestId !== creationParameters.previousRequestId)
+      ) {
+        throw Error(
+          'recurrenceNumber must be 1 if masterRequestId and previousRequestId are equal and vice versa',
+        );
+      }
+
+      return {
+        action: 'create',
+        id: this.extensionId,
+        parameters: {
+          previousRequestId: creationParameters.previousRequestId,
+          masterRequestId: creationParameters.masterRequestId,
+          recurrenceNumber: creationParameters.recurrenceNumber,
+        },
+        version: this.currentVersion,
+      } as ExtensionTypes.IAction<TCreationParameters>;
     }
 
-    return super.createCreationAction(
-      creationParameters,
-    ) as ExtensionTypes.IAction<TCreationParameters>;
+    throw Error(
+      'masterRequestId, previousRequestId and recurrenceNumber must be all empty or all filled',
+    );
   }
 
+  /**
+   * Applies a creation extension action
+   *
+   * @param extensionAction action to apply
+   * @param timestamp ?
+   *
+   * @returns state of the extension created
+   */
   protected applyCreation(
     extensionAction: ExtensionTypes.IAction,
     timestamp: number,
   ): ExtensionTypes.IState {
-    if (!extensionAction.parameters.expectedStartDate) {
-      throw Error('expectedStartDate should not be empty');
-    }
-    if (!extensionAction.parameters.expectedFlowRate) {
-      throw Error('expectedFlowRate should not be empty');
-    }
+    /* Master request Creation */
     if (
-      extensionAction.parameters.expectedStartDate &&
-      !Utils.amount.isValid(extensionAction.parameters.expectedStartDate)
+      !(
+        extensionAction.parameters.masterRequestId ||
+        extensionAction.parameters.previousRequestId ||
+        extensionAction.parameters.recurrenceNumber
+      )
     ) {
-      throw Error('expectedStartDate is not a valid amount');
+      if (!extensionAction.parameters.expectedStartDate) {
+        throw Error('expectedStartDate should not be empty');
+      }
+      if (!extensionAction.parameters.expectedFlowRate) {
+        throw Error('expectedFlowRate should not be empty');
+      }
```

Comment:
Is it OK to check both the presence and the validity in one condition? It's easier to read and maintain